znvis/visualizer/visualizer.py

```diff
@@ -95,14 +95,19 @@ def __init__(
         self.bounding_box = bounding_box() if bounding_box else None
 
         if number_of_steps is None:
-            number_of_steps = len(particles[0].position)
+            len_list = []
+            for particle in particles:
+                if not particle.static:
+                    len_list.append(len(particle.position))
+            number_of_steps = min(len_list)
```

Comment:
This will break for static meshes as they have a length of 1.

Reply:
Thanks! I did miss the case where only static meshes are to be visualized. I'll add a check for this.

I still think taking the min is necessary, since if I visualized two particle trajectories of which one has less data points than the other, at some point an index out of range error would be thrown if I used max.